### PR TITLE
[Logic] Request Access Token: Certificate Pinning (EXPOSUREAPP-10706)

### PR DESCRIPTION
**Pull request by jurajkusnier**
### Requesting the Access Token
- implement cert pinning
- add unit tests
- updates according to the latest specs

### Jira Ticket
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-10706

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/service/processor/AccessTokenRequestProcessor.kt [hunk 1]
```diff
@@ -60,29 +61,47 @@ class AccessTokenRequestProcessor @Inject constructor(
...
-            else -> DccTicketingErrorCode.VS_ID_SERVER_ERR
+            else -> DccTicketingErrorCode.ATR_SERVER_ERR
         }.let { DccTicketingException(it, e) }
     }
 
-    private fun verifyJWT(jwt: String, jwkSet: Set<DccJWK>) {
+    private fun verifyJWT(jwt: String, jwkSet: Set<DccJWK>) = try {
         jwtVerification.verify(jwt, jwkSet)
+    } catch (e: DccTicketingJwtException) {
+        Timber.e(e, "verifyJWT for result token failed")
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccticketing/core/service/processor/AccessTokenRequestProcessor.kt:88]: for access token …

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/dccticketing/core/service/processor/ResultTokenRequestProcessorTest.kt [hunk 2]
```diff
@@ -136,10 +136,10 @@ internal class ResultTokenRequestProcessorTest : BaseTest() {
     @Test
     fun `verifyJWT throws RTR_JWT_VER_NO_JWKS`() = runBlockingTest {
```
Comment (mtwalli) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/dccticketing/core/service/processor/ResultTokenRequestProcessorTest.kt:137]: Please update the error code in the name ?
RTR_JWT_VER_EMPTY_JWKS

File: Corona-Warn-App/src/test/java/de/rki/coronawarnapp/dccticketing/core/service/processor/AccessTokenRequestProcessorTest.kt [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+            )
+        } throws NetworkConnectTimeoutException()
+
+        shouldThrow<DccTicketingException> {
+            callRequestAccessToken()
+        }.errorCode shouldBe DccTicketingException.ErrorCode.ATR_NO_NETWORK
+    }
+
+    @Test
+    fun `verifyJWT throws ATR_JWT_VER_NO_JWKS`() = runBlockingTest {
```
Comment (mtwalli) [inline Corona-Warn-App/src/test/java/de/rki/coronawarnapp/dccticketing/core/service/processor/AccessTokenRequestProcessorTest.kt:122]: error code should be ATR_JWT_VER_EMPTY_JWKS in the name

**Review by mtwalli — COMMENTED** [review]
LGTM, just minor typos

**Review by mtwalli — APPROVED** [review]

**Review by BMItr — APPROVED** [review]
lgtm 👌

**Pull request merged**